### PR TITLE
Add access control, add health check on port 81

### PR DESCRIPTION
**Pull request by etiennetremel**
* add access control
* add health check on port 81
* add extra resolver 8.8.4.4

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -24,16 +24,12 @@ ADD ssl/ssl.pem /etc/ssl/private/ssl.pem
...
 
 # expose ports
-EXPOSE 80 443
+EXPOSE 80 81 443
 
 # runtime environment variables
 ENV BACKEND_SERVER=localhost \
-    BACKEND_SERVER_PORT=80
+    BACKEND_SERVER_PORT=80 \
+    WHITELIST_CIDRS=""
```
Comment (JorritSalverda) [inline Dockerfile]: Does it work with an empty string or should it be set to 0.0.0.0/0 like we do for haproxy?

File: Dockerfile [hunk 2]
```diff
@@ -24,16 +24,12 @@ ADD ssl/ssl.pem /etc/ssl/private/ssl.pem
...
 
-# define default command
-CMD deluser $USER_NAME; \
-    addgroup -S -g $GROUP_ID $GROUP_NAME; \
-    adduser -S -G $GROUP_NAME -u $USER_ID $USER_NAME; \
-    chown $USER_NAME:$GROUP_NAME -R /tmp/nginx/client_body_temp /tmp/nginx/proxy_temp; \
-    sed -i -e "s/localhost:80/${BACKEND_SERVER}:${BACKEND_SERVER_PORT}/" /etc/nginx/nginx.conf; \
```
Comment (JorritSalverda) [inline Dockerfile]: We still need these replacements. The user stuff is probably fine being left out.

File: entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
```
Comment (JorritSalverda) [inline entrypoint.sh]: Ah, here's the stuff from CMD. However I don't see the entrypoint.sh being executed anywhere.

**Comment by etiennetremel** [comment]
Yes, I entrypoint is now there, forgot to add it to the Docker file

File: nginx.conf [hunk 4]
```diff
@@ -93,17 +96,31 @@ http {
...
     }
-}
\ No newline at end of file
+
+    # health
+    server {
+        listen       [::]:81;
+        listen       81;
+
+        location /healthz {
```
Comment (JorritSalverda) [inline nginx.conf]: Better call this /liveness in line with kubernetes. 

/readiness depends on the backends being able to accept traffic.

**Pull request merged**